### PR TITLE
Implement support for now stable rename_dependency feature. Fixes #133.

### LINKED ISSUE #133: Use correct name when dependency has been renamed
Cargo [recently](https://github.com/rust-lang/cargo/blob/0dd53841d30816ec0d79d879de71d7adc360bdc5/src/doc/src/reference/unstable.md#rename-dependency) got support for renaming dependencies. That is, you can do:
```toml
cargo-features = ["rename-dependency"]
# ...
[dependencies]
bar = { version = "0.1", package = "foo" }
```
To import the crate `foo` under the name `bar`. However, this seems to confuse `cargo-outdated`. Specifically, it looks for `bar` instead of `foo` in the registry, which generally won't exist.

### PR DESCRIPTION
**Pull request by leoschwarz**
This is a minor patch, but with a project of mine I still experience an error (version conflict) so I'm not certain if it really fixes the issue yet. I'll have to investigate that further.

### CODE REVIEW

File: src/cargo_ops/temp_project.rs [hunk 1]
```diff
@@ -394,6 +396,13 @@ impl<'tmp> TempProject<'tmp> {
                     }
                 }
                 Value::Table(ref t) => {
+                    let name = match t.get("package") {
+                        Some(&Value::String(ref s)) => s,
+                        // TODO: Probably this should emit an error?
```
Comment (leoschwarz) [inline src/cargo_ops/temp_project.rs]: As far as I understand "package" for a dependency if present can only be a string indicating the name of the real crates.io package. Right now a "package" value which is not a string is silently ignored, but maybe I should turn that into an error too?

File: src/cargo_ops/temp_project.rs [hunk 2]
```diff
@@ -394,6 +396,13 @@ impl<'tmp> TempProject<'tmp> {
...
+                        Some(_) => &dep_key,
```
Comment (leoschwarz) [inline src/cargo_ops/temp_project.rs]: As far as I understand "package" for a dependency if present can only be a string indicating the name of the real crates.io package. Right now a "package" value which is not a string is silently ignored, but maybe I should turn that into an error too?

**Comment by alexheretic** [comment]
Works for me thanks! Fixes #133

**Comment by kornelski** [comment]
Thanks!

File: src/cargo_ops/temp_project.rs (hunk 2 above)
Comment (kornelski) [reply src/cargo_ops/temp_project.rs]: Cargo rejects non-strings here, so it'd be OK to reject them too.

**Pull request merged**